src/tir/schedule/primitive/get_block_loop.cc

```diff
@@ -80,27 +80,27 @@ Array<StmtSRef> GetChildBlocks(const ScheduleState& self, const StmtSRef& parent
 Array<StmtSRef> GetProducers(const ScheduleState& self, const StmtSRef& block_sref) {
   StmtSRef scope_root = GetScopeRoot(self, block_sref, /*require_stage_pipeline=*/false);
   Array<Dependency> edges = self->GetBlockScope(scope_root)->GetDepsByDst(block_sref);
-  Array<StmtSRef> results;
+  std::unordered_set<StmtSRef, ObjectPtrHash, ObjectPtrEqual> results;
   results.reserve(edges.size());
   for (const Dependency& edge : edges) {
     if (edge->kind == DepKind::kRAW || edge->kind == DepKind::kWAW) {
-      results.push_back(edge->src);
+      results.emplace(edge->src);
     }
   }
-  return results;
+  return Array<StmtSRef>(results.begin(), results.end());
```

Comment:
Note: `std::unordered_set` doesn't guarantee proper ordering, which makes scheduling traces not strictly reproducible